full-detector-simulations/FCCeeCLD/FCCeeCLD.md

```diff
@@ -23,3 +23,678 @@ source /cvmfs/sw-nightlies.hsf.org/key4hep/setup.sh
+    ROOT.gGeoManager.SetVisOption(0)
+    ROOT.gGeoManager.Export(out_path)
+
+
+if __name__ == '__main__':
+    main()
+
+```
+
+</details>
+
+<br/>
+
+2. Using Qt interface of Geant4, accesible via `runType` argument of `ddsim`
+
+<details>
+
+<summary><b> Click to show how to display the detector geometry using Geant4. </b></summary>
+
+```shell
+source /cvmfs/sft.cern.ch/lcg/views/dev4/latest/x86_64-centos7-gcc11-opt/setup.sh
+ddsim --compactFile description.xml --runType qt
+```
+
+After building internally the geometry, a window will open. The menu on the left gives access to the builtin Geant4 macros. In order to show the geometry, we have to type some commands:
+
+```
+# Use this open statement to create an OpenGL view:
+/vis/open OGL 600x600-0+0
+
+# Draw geometry:
+/vis/drawVolume
+
+# Specify view angle:
+/vis/viewer/set/lightsVector -1 0 0
+/vis/viewer/set/viewpointVector -1 0 0
+```
+
+
+</details>
+:::
+
+<br/>
+
+
+### Simulate Muons in CLD detector
+
+To simulate the interaction of 100 muons with the CLD detector, the following command is used inside the directory `mydd4heptutorial`. It takes some minutes to build the geometry and run the actual simulation. 
+
+```shell
+ddsim --compactFile FCCee_o1_v05/FCCee_o1_v05.xml \
+      --enableGun \
+      --gun.distribution uniform \
+      --gun.energy "10*GeV" \
+      --gun.particle mu- \
+      --numberOfEvents 100 \
+      --outputFile Step1_edm4hep.root
+```
+
+And now plot the distribution with `showPlots.py`
+
+```shell
+python showPlots.py Step1_edm4hep.root
+```
+
+### Removing an included XML file
+
+The detector geometry is complex, and takes a while to be built internally. To speed up the following steps, the Silicon Tracker can be removed. To do so, 
+
+1. Open `FCCee_o1_v05/FCCee_o1_v05.xml`
+2. Find the following lines in the file, and replace/delete them
+
+```xml
+<include ref="InnerTracker_o2_v06_02.xml"/>
+<include ref="OuterTracker_o2_v06_02.xml"/>
+```
+
+3. Comment out the plugins section `<plugins> ... </plugins>` in the same file. 
+
+If now the simulation is run again as in the previous step (changing the output filename), 
+
+```shell
+ddsim --compactFile FCCee_o1_v05/FCCee_o1_v05.xml \
+      --enableGun \
+      --gun.distribution uniform \
+      --gun.energy "10*GeV" \
+      --gun.particle mu- \
+      --numberOfEvents 100 \
+      --outputFile Step2_edm4hep.root
+```
+
+The simulation runs 20 times faster and the distributions have not noticeably changed.
+
+### Overlap checking
+
+Whenever you change the geometry in a non-trivial way there are the possibilities of overlaps and the following things should be kept in mind
+
+1. There are no trivial changes
+2. See point 1
+3. Run the overlap check
+
+Geant4 performs the overlap check of a given geometry. This action is available for `ddsim`. A a macro file with the Geant4 instructions has to be provided to `ddsim`. To do so, create a file named as *overlap.mac* and write these commands inside
```

Comment:
```suggestion
Geant4 can perform the overlap check of a given geometry. This action is available for `ddsim`. A macro file with the Geant4 instructions has to be provided to `ddsim`. To do so, create a file named as *overlap.mac* and write these commands inside
```
One can also give the commands manually, but if the output is too long it will be lost, so we use a macro file so that we can redirect the output and end ddsim gracefully at the end.